addons/infinite-scroll/buttonScroll.js

```diff
@@ -17,11 +18,11 @@ async function commentLoader(addon, heightControl, selector, pathname, { yProvid
     prevScrollDetector = scrollDetecter;
     func = () => {
       const threshold = yProvider ? yProvider.scrollTop + yProvider.clientHeight : window.scrollY + window.innerHeight;
+      let height = el.closest(heightControl).offsetHeight;
       if (typeof pathname === "string" && (window.location.pathname.split("/")[3] || "") !== pathname) return;
-      if (threshold >= el.closest(heightControl).offsetHeight - 500) {
-        if (el) {
-          el.click();
-        }
+      if (threshold >= height - 500 && height > prevHeight) {
+        prevHeight = height;
+        el.click();
```

Comment:
@Jazza-231 I don't think this actually fixes double-clicking, if the user is clicking slowly. The best solution would be to avoid clicking the button while the page is loading.